File: mobile_app/lib/models/DeliveryAgent.dart
import 'package:mobile_app/models/Pickup.dart';
import 'package:mobile_app/models/Shipment.dart';

class DeliveryAgent {
  final int? id;
  final int? phoneNumber;
  final String? name;
  final String? role;
  final List<Pickup>? pickups;
  final List<Shipment>? shipments;

  DeliveryAgent({
    required this.id,
    required this.name,
    required this.phoneNumber,
    required this.role,
    this.shipments,
    this.pickups,
  });

  factory DeliveryAgent.fromJson(json) {
    return DeliveryAgent(
      id: json['id'],
      name: json['name'],
      phoneNumber: json['phoneNumber'],
      role: json['role'],
      shipments: json['shipments'],
      pickups: json['pickups'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'phoneNumber': phoneNumber,
      'role': role,
      'shipments': shipments,
      'pickups': pickups
    };
  }
}

File: demo/src/main/java/com/backend/demo/entity/DeliveryAgent.java
package com.backend.demo.entity;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;

@Entity
@Table(name="delivery_agent")
public class DeliveryAgent {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id")
	private int id;
	@Column(name="name")
	private String name;
	@Column(name="phone_number")
	private int phoneNumber;
	@Column(name = "password")
	@JsonIgnore
	private String password;
	
	@OneToMany(mappedBy = "deliveryAgent" , cascade = {CascadeType.REFRESH,CascadeType.DETACH,CascadeType.MERGE,CascadeType.PERSIST,CascadeType.REFRESH} )
	@JsonIgnore
	private List<Shipment> shipments;
	
	@OneToMany(mappedBy = "deliveryAgent" , cascade = {CascadeType.REFRESH,CascadeType.DETACH,CascadeType.MERGE,CascadeType.PERSIST,CascadeType.REFRESH} )
	@JsonIgnore
	private List<PickUp> pickups;
	
	@Column
	private String role;
	
	public DeliveryAgent() {}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public int getPhoneNumber() {
		return phoneNumber;
	}

	public void setPhoneNumber(int phoneNumber) {
		this.phoneNumber = phoneNumber;
	}

	public String getPassword() {
		return password;
	}

	public void setPassword(String password) {
		this.password = password;
	}

	public List<Shipment> getShipments() {
		return shipments;
	}

	public void setShipments(List<Shipment> shipments) {
		this.shipments = shipments;
	}

	public String getRole() {
		return role;
	}

	public void setRole(String role) {
		this.role = role;
	}

	public List<PickUp> getPickups() {
		return pickups;
	}

	public void setPickups(List<PickUp> pickups) {
		this.pickups = pickups;
	}
	
	
	
}
